darglint2/parse/grammars/sphinx_arguments_section.py

```diff
@@ -84,7 +84,7 @@ class ArgumentsGrammar(BaseGrammar):
         P("item-body7", ([], "newline", "block", 0)),
         P("block-indented0", ([], "split", "block-indented", 0)),
         P("block-indented1", ([], "split-indented", "block-indented", 0)),
-        P("split-indented0", ([], "indents", "newlines", 0), ([], "indent", "indents", 0), (TokenType.INDENT, 0), ([], "newline", "newlines", 0), (TokenType.NEWLINE, 0)),
+        P("split-indented0", ([], "indents", "newlines", 0), ([], "newline", "newlines", 0), (TokenType.NEWLINE, 0), ([], "indent", "indents", 0), (TokenType.INDENT, 0)),
```

Comment:
@terrencepreilly I wonder why I see these kinds of differences when regenerating the grammars using `bnf_to_cnf`. Any idea? Do these look like they might change the behavior?

Reply:
It's just a change in the order of the patterns.  It comes from the production

```
<item-body>
    ::= 2 <line> <newline> <paragraph-indented-two>
    | 2 <line> <newline> <paragraph-indented-two> <newline>
    | 2 <line> <newline>
    | 2 <line>
    | @IndentError
      <line> <newline> <paragraph>
    | @IndentError
      <line> <newline> <paragraph> <newline>
```

Typically, the order does matter when you're generating something like an LR parser or something.  But since this is using CYK, it doesn't actually matter, since it will identify all possible parse trees.

If you change it to use a different algorithm to make it faster or something (like I was working on before I gave up on the project: see the parser-generator branch), then it could probably be an issue.  Currently, it doesn't matter.

Reply:
So what algorithm were you trying to implement @terrencepreilly?

Reply:
Oh, I was writing an LL(K) parser generator.  The idea was that we could use the same (or almost the same) grammar definition for both the CYK parser and the LL(K) parser.  When the LL(K) parser failed to parse a docstring (due to some ambiguity), it would fall back to the CYK parser.

Reply:
Thanks!